src/midiEditorCommands.cpp

```diff
@@ -357,6 +372,74 @@ vector<MidiNote> getSelectedNotes(MediaItem_Take* take, int offset=-1) {
 	return notes;
 }
 
+// Finds a single CC at the cursor in a given direction and returns its info.
+// This updates currentCC.
+MidiControlChange findCC(MediaItem_Take* take, int direction) {
+	int count;
+	MIDI_CountEvts(take, NULL, &count, NULL);
+	double cursor = GetCursorPosition();
+	double position;
+	int start = direction == -1 ? count - 1 : 0;
+	if (currentCC != -1) {
+		MIDI_GetCC(take, currentCC, NULL, NULL, &position, NULL, NULL, NULL, NULL);
+		position = MIDI_GetProjTimeFromPPQPos(take, position);
+		if (direction == 0 && position == cursor) {
+			start = currentCC;
+		} else if (direction == 1 ? position <= cursor : position >= cursor) {
+			// The cursor is right at or has moved past the CC to which the user last moved.
+			// Therefore, start at the adjacent CC.
+			start = currentCC + direction;
+			if (start < 0 || start >= count) {
+				// There's no adja	cent item in this direction,
+				// so move to the current one again.
+				start = currentCC;
+			}
+		}
+	}
+	bool found = false;
+	int chan, control, value;
+	for (; 0 <= start && start < count; start += direction) {
+		MIDI_GetCC(take, start, NULL, NULL, &position, NULL, &chan, &control, &value);
+		position = MIDI_GetProjTimeFromPPQPos(take, position);
+		if (direction == -1 ? position <= cursor : position >= cursor) {
+			currentCC = start;
+			found = true;
+			break;
+		}
+	}
+	if (!found) {
+		return {-1};
+	}
+	return {chan, currentCC, control, value, position};
+}
+
+void selectCC(MediaItem_Take* take, const int cc, bool select=true) {
+	MIDI_SetCC(take, cc, &select, NULL, NULL, NULL, NULL, NULL, NULL, NULL);
+}
+
+bool isCCSelected(MediaItem_Take* take, const int cc) {
+	bool sel;
+	MIDI_GetCC(take, cc, &sel, NULL, NULL, NULL, NULL, NULL, NULL);
+	return sel;
+}
+
+vector<MidiControlChange> getSelectedCCs(MediaItem_Take* take, int offset=-1) {
```

Comment:
1. This doesn't seem to be used in this PR. Can we move it into another?
2. What's the use case for the offset argument? I just saw that getSelectedNotes has it as well, though I didn't think to ask this question in that review. 😳

Reply:
> This doesn't seem to be used in this PR. Can we move it into another?

Sure, will probably need it in a subsequent pr but will introduce it there.

> What's the use case for the offset argument? I just saw that getSelectedNotes has it as well, though I didn't think to ask this question in that review. 😳

It allows you to start searching for selected notes from a certain note, leaving notes before out of scope. I'm currently not sure about the use case though, I just decided to expose it as an optional parameter instead of a constant.